extensions/amp-carousel/0.2/amp-carousel.js

```diff
@@ -153,6 +153,8 @@ class AmpCarousel extends AMP.BaseElement {
     this.prevButton_ = this.element.querySelector('.amp-carousel-button-prev');
     this.nextButton_ = this.element.querySelector('.amp-carousel-button-next');
 
+    this.applyFillContent(this.scrollContainer_);
```

Comment:
I'm surprised we'd want to apply `max-width: 100%` to `scrollContainer_` - shouldn't this be the width of 100% x numSlides? When I apply `i-amphtml-fill-content` class to the scroll element manually to the [glitch](https://amp-demos.glitch.me/amp-carousel.html), I'm not able to scroll horizontally between the slide items / it looks like they are on top of each other. Should I be testing a different way?

Reply:
I do not understand how carousel works here. Why are the slides on top of each other?

Reply:
It's worrying that this completely broke, but no tests failed.

Reply:
I think the real issue is that `fill` children just fundamentally don't work within a carousel.

Reply:
It looks like the slides are on top of each other instead of next to each other in a larger scrollable container because the outer container has `i-amphtml-fill-content` and the slides each do as well... I don't think that is the intended behavior here. I agree, `fill` children doesn't make sense in a carousel that should be scrollable? When did this used to work?

Reply:
I don't know if it ever did. @sebastianbenz

Reply:
Not sure either. I always use fill layout for carousel children when testing, but never really thought about it. Might be that I usually combined it with type=slides. But thinking about it, I agree with Caroline that it doesn't really make sense.

Reply:
It does look like `amp-carousel[type=carousel][layout=responsive]` with `layout=responsive` children is broken due to `position:static` causing the carousel to layout below the sizer element--[repro](https://playground.amp.dev/#share=PCFkb2N0eXBlIGh0bWw+CjxodG1sIOKaoSBsYW5nPSJlbiI+CjxoZWFkPgogIDxtZXRhIGNoYXJzZXQ9InV0Zi04Ij4KICA8dGl0bGU+YW1wLWFjY29yZGlvbjwvdGl0bGU+CiAgPHNjcmlwdCBhc3luYyBzcmM9Imh0dHBzOi8vY2RuLmFtcHByb2plY3Qub3JnL3YwLmpzIj48L3NjcmlwdD4KICA8c2NyaXB0IGFzeW5jIGN1c3RvbS1lbGVtZW50PSJhbXAtY2Fyb3VzZWwiIHNyYz0iaHR0cHM6Ly9jZG4uYW1wcHJvamVjdC5vcmcvdjAvYW1wLWNhcm91c2VsLTAuMi5qcyI+PC9zY3JpcHQ+CiAgPHNjcmlwdCBhc3luYyBjdXN0b20tZWxlbWVudD0iYW1wLWxpZ2h0Ym94LWdhbGxlcnkiIHNyYz0iaHR0cHM6Ly9jZG4uYW1wcHJvamVjdC5vcmcvdjAvYW1wLWxpZ2h0Ym94LWdhbGxlcnktMC4xLmpzIj48L3NjcmlwdD4KICA8bGluayByZWw9ImNhbm9uaWNhbCIgaHJlZj0iLiI+CiAgPG1ldGEgbmFtZT0idmlld3BvcnQiIGNvbnRlbnQ9IndpZHRoPWRldmljZS13aWR0aCI+CiAgPHN0eWxlIGFtcC1ib2lsZXJwbGF0ZT4gYm9keSB7IC13ZWJraXQtYW5pbWF0aW9uOiAtYW1wLXN0YXJ0IDhzIHN0ZXBzKDEsIGVuZCkgMHMgMSBub3JtYWwgYm90aDsgLW1vei1hbmltYXRpb246IC1hbXAtc3RhcnQgOHMgc3RlcHMoMSwgZW5kKSAwcyAxIG5vcm1hbCBib3RoOyAtbXMtYW5pbWF0aW9uOiAtYW1wLXN0YXJ0IDhzIHN0ZXBzKDEsIGVuZCkgMHMgMSBub3JtYWwgYm90aDsgYW5pbWF0aW9uOiAtYW1wLXN0YXJ0IDhzIHN0ZXBzKDEsIGVuZCkgMHMgMSBub3JtYWwgYm90aCB9IEAtd2Via2l0LWtleWZyYW1lcyAtYW1wLXN0YXJ0IHsgZnJvbSB7IHZpc2liaWxpdHk6IGhpZGRlbiB9IHRvIHsgdmlzaWJpbGl0eTogdmlzaWJsZSB9IH0gQC1tb3ota2V5ZnJhbWVzIC1hbXAtc3RhcnQgeyBmcm9tIHsgdmlzaWJpbGl0eTogaGlkZGVuIH0gdG8geyB2aXNpYmlsaXR5OiB2aXNpYmxlIH0gfSBALW1zLWtleWZyYW1lcyAtYW1wLXN0YXJ0IHsgZnJvbSB7IHZpc2liaWxpdHk6IGhpZGRlbiB9IHRvIHsgdmlzaWJpbGl0eTogdmlzaWJsZSB9IH0gQC1vLWtleWZyYW1lcyAtYW1wLXN0YXJ0IHsgZnJvbSB7IHZpc2liaWxpdHk6IGhpZGRlbiB9IHRvIHsgdmlzaWJpbGl0eTogdmlzaWJsZSB9IH0gQGtleWZyYW1lcyAtYW1wLXN0YXJ0IHsgZnJvbSB7IHZpc2liaWxpdHk6IGhpZGRlbiB9IHRvIHsgdmlzaWJpbGl0eTogdmlzaWJsZSB9IH0gPC9zdHlsZT48bm9zY3JpcHQ+IDxzdHlsZSBhbXAtYm9pbGVycGxhdGU+IGJvZHkgeyAtd2Via2l0LWFuaW1hdGlvbjogbm9uZTsgLW1vei1hbmltYXRpb246IG5vbmU7IC1tcy1hbmltYXRpb246IG5vbmU7IGFuaW1hdGlvbjogbm9uZSB9IDwvc3R5bGU+IDwvbm9zY3JpcHQ+CjwvaGVhZD4KPGJvZHk+CjxhbXAtY2Fyb3VzZWwgd2lkdGg9IjQwMCIgaGVpZ2h0PSIzMDAiIGxheW91dD0icmVzcG9uc2l2ZSIgdHlwZT0iY2Fyb3VzZWwiIHJvbGU9InJlZ2lvbiIgYXJpYS1sYWJlbD0iQmFzaWMgdXNhZ2UgY2Fyb3VzZWwiPgogIDxhbXAtaW1nIHNyYz0iaHR0cHM6Ly9hbXAuZGV2L3N0YXRpYy9zYW1wbGVzL2ltZy9pbWFnZTEuanBnIiB3aWR0aD0iNDAwIiBoZWlnaHQ9IjMwMCIgYWx0PSJhIHNhbXBsZSBpbWFnZSIgbGF5b3V0PSJmaXhlZCI+PC9hbXAtaW1nPgogIDxhbXAtaW1nIHNyYz0iaHR0cHM6Ly9hbXAuZGV2L3N0YXRpYy9zYW1wbGVzL2ltZy9pbWFnZTIuanBnIiB3aWR0aD0iNDAwIiBoZWlnaHQ9IjMwMCIgYWx0PSJhbm90aGVyIHNhbXBsZSBpbWFnZSIgbGF5b3V0PSJmaXhlZCI+PC9hbXAtaW1nPgogIDxhbXAtaW1nIHNyYz0iaHR0cHM6Ly9hbXAuZGV2L3N0YXRpYy9zYW1wbGVzL2ltZy9pbWFnZTMuanBnIiB3aWR0aD0iNDAwIiBoZWlnaHQ9IjMwMCIgYWx0PSJhbmQgYW5vdGhlciBzYW1wbGUgaW1hZ2UiIGxheW91dD0iZml4ZWQiPjwvYW1wLWltZz4KPC9hbXAtY2Fyb3VzZWw+CjwvYm9keT4KPC9odG1sPg==). 

I would say this PR does not particularly help the `fill` children case (images that weren't loading now load, but the carousel still does not fundamentally work, i.e. you can't scroll in it). But this PR _does_ appear to help the `responsive` case (particularly removing `position: static` works, and applying fill content is not harmful), though I am surprised this hasn't been reported as broken, especially since `position: static` has been in the carousel style since it was created. Could people potentially be relying on this "broken" behavior? Would "fixing" it cause publishers to be surprised by the change?